internal/plugin/plugin_test.go

```diff
@@ -94,10 +94,188 @@ func TestHTTPReqCall_FailedToParseConf(t *testing.T) {
+			args: args{
+				name: "1",
+				pc:   nil,
+				sv:   nil,
+			},
+			wantErr: ErrMissingParseConfMethod,
+		},
+		{
+			name: "test_MissingName&ParseConfMethod",
+			args: args{
+				name: "",
+				pc:   nil,
+				sv:   emptyFilter,
+			},
+			wantErr: ErrMissingName,
+		},
+		{
+			name: "test_MissingName&FilterMethod",
+			args: args{
+				name: "",
+				pc:   emptyParseConf,
+				sv:   nil,
+			},
+			wantErr: ErrMissingName,
+		},
+		{
+			name: "test_MissingAll",
+			args: args{
+				name: "",
+				pc:   nil,
+				sv:   nil,
+			},
+			wantErr: ErrMissingName,
+		},
+		{
+			name: "test_plugin1",
+			args: args{
+				name: "plugin1",
+				pc:   emptyParseConf,
+				sv:   emptyFilter,
+			},
+			wantErr: nil,
+		},
+		{
+			name: "test_plugin1_again",
+			args: args{
+				name: "plugin1",
+				pc:   emptyParseConf,
+				sv:   emptyFilter,
+			},
+			wantErr: ErrPluginRegistered{"plugin1"},
+		},
+		{
+			name: "test_plugin2",
+			args: args{
+				name: "plugin2111%#@#",
+				pc:   emptyParseConf,
+				sv:   emptyFilter,
+			},
+			wantErr: nil,
+		},
+		{
+			name: "test_plugin3",
+			args: args{
+				name: "plugin311*%#@#",
+				pc:   emptyParseConf,
+				sv:   emptyFilter,
+			},
+			wantErr: nil,
+		},
+		{
+			name: "test_plugin3_again",
+			args: args{
+				name: "plugin311*%#@#",
+				pc:   emptyParseConf,
+				sv:   emptyFilter,
+			},
+			wantErr: ErrPluginRegistered{"plugin311*%#@#"},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			if err := RegisterPlugin(tt.args.name, tt.args.pc, tt.args.sv); !assert.Equal(t, tt.wantErr, err) {
+				t.Errorf("RegisterPlugin() error = %v, wantErr %v", err, tt.wantErr)
+			}
+		})
+	}
+}
+
+func TestRegisterPluginConcurrent(t *testing.T) {
+	RegisterPlugin("test_concurrent-1", emptyParseConf, emptyFilter)
+	RegisterPlugin("test_concurrent-2", emptyParseConf, emptyFilter)
+	type args struct {
+		name string
+		pc   ParseConfFunc
+		sv   FilterFunc
+	}
+	type test struct {
+		name    string
+		args    args
+		wantErr error
+	}
+	tests := []test{
+		{
+			name: "test_concurrent-1",
+			args: args{
+				name: "test_concurrent-1",
+				pc:   emptyParseConf,
+				sv:   emptyFilter,
+			},
+			wantErr: ErrPluginRegistered{"test_concurrent-1"},
+		},
+		{
+			name: "test_concurrent-2#01",
+			args: args{
+				name: "test_concurrent-2",
+				pc:   emptyParseConf,
+				sv:   emptyFilter,
+			},
+			wantErr: ErrPluginRegistered{"test_concurrent-2"},
+		},
+		{
+			name: "test_concurrent-2#02",
+			args: args{
+				name: "test_concurrent-2",
+				pc:   emptyParseConf,
+				sv:   emptyFilter,
+			},
+			wantErr: ErrPluginRegistered{"test_concurrent-2"},
+		},
+		{
+			name: "test_concurrent-2#03",
+			args: args{
+				name: "test_concurrent-2",
+				pc:   emptyParseConf,
+				sv:   emptyFilter,
+			},
+			wantErr: ErrPluginRegistered{"test_concurrent-2"},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			for i := 0; i < 10000; i++ {
```

Comment:
LGTM, but `10000` is too high to me. I think 3 is enough.

Reply:
thanks .  
yes , it  is .   i changed it  to 3